internal/config/main.go

```diff
@@ -197,6 +206,35 @@ func (c *Config) RegisterOpenShiftRoutesChangeCallback(f func() error) {
 	c.onOpenShiftRoutesChange.Register(f)
 }
 
+type hpaVersionStore interface {
+	Set(hpaV autodetect.AutoscalingVersion)
+	Get() autodetect.AutoscalingVersion
+}
+
+func newHPAVersionWrapper() hpaVersionStore {
+	return &hpaVersionWrapper{
+		current: autodetect.AutoscalingVersionUnknown,
+	}
+}
+
+type hpaVersionWrapper struct {
+	mu      sync.Mutex
+	current autodetect.AutoscalingVersion
+}
+
+func (p *hpaVersionWrapper) Set(hpaV autodetect.AutoscalingVersion) {
```

Comment:
to confirm the change here is that we're passing this to copy the value and using a lock to ensure there's no contention?